<launch>
  <arg name="node_name" default="line_segment_detector" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />

  <arg name="lsd_refine_type" default="2" doc="Line Segment Detector Modes. 0: No refinement applied, 1: Standard refinement is applied, 2, Advanced refinement is applied."/>
  <arg name="lsd_scale" default="0.8" doc="The scale of the image that will be used to find the lines. Range (0..1]" />
  <arg name="lsd_sigma_scale" default="0.6" doc="Sigma for Gaussian filter. It is computed as sigma = _sigma_scale/_scale" />
  <arg name="lsd_quant" default="2.0" doc="Bound to the quantization error on the gradient norm" />
  <arg name="lsd_angle_threshold" default="22.5" doc="Gradient angle tolerance in degrees" />
  <arg name="lsd_log_eps" default="0" doc="Detection threshold: -log10(NFA) > log_eps. Used only when advancent refinement is chosen" />
  <arg name="lsd_density_threshold" default="0.3" doc="Minimal density of aligned region points in the enclosing rectangle" />
  <arg name="lsd_n_bins" default="1024" doc="Number of bins in pseudo-ordering of gradient modulus" />
  <arg name="lsd_line_length_threshold" default="0.0" doc="Threshold of line length" />

  <!-- line_segment_detector.cpp  -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="line_segment_detector" >
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="lsd_refine_type" value="$(arg lsd_refine_type)" />
    <param name="lsd_scale" value="$(arg lsd_scale)" />
    <param name="lsd_sigma_scale" value="$(arg lsd_sigma_scale)" />
    <param name="lsd_quant" value="$(arg lsd_quant)" />
    <param name="lsd_angle_threshold" value="$(arg lsd_angle_threshold)" />
    <param name="lsd_log_eps" value="$(arg lsd_log_eps)" />
    <param name="lsd_density_threshold" value="$(arg lsd_density_threshold)" />
    <param name="lsd_n_bins" value="$(arg lsd_n_bins)" />
    <param name="lsd_line_length_threshold" value="$(arg lsd_line_length_threshold)" />
  </node>
</launch>
